src/include/mpir_misc.h

```diff
@@ -89,4 +89,16 @@ int MPIR_Find_external(struct MPIR_Comm *comm, int *external_size_p, int *extern
 int MPIR_Get_internode_rank(MPIR_Comm * comm_ptr, int r);
 int MPIR_Get_intranode_rank(MPIR_Comm * comm_ptr, int r);
 
+#define MPIR_CAST(T, val) CAST_##T((val))
+#ifdef NDEBUG
+#define MPIR_CAST_int(val) ((int) (val))
+#define MPIR_CAST_Aint(val) ((MPI_Aint) (val))
+#else
+#define MPIR_CAST_int(val) \
+    (((val) > INT_MAX || ((val) < 0 && (val) < INT_MIN)) ? (assert(0), 0) : (int) (val))
+#define MPIR_CAST_Aint(val) \
+    (((val) > MPIR_AINT_MAX || ((val) < 0 && (val) < MPIR_AINT_MIN)) ? (assert(0), 0) \
```

Comment:
Similarly.

Reply:
Refer to https://github.com/pmodels/mpich/pull/6145#discussion_r966589870

Reply:
Also, please fix the commit title since it is no longer in `mpl`

Reply:
> Refer to [#6145 (comment)](https://github.com/pmodels/mpich/pull/6145#discussion_r966589870)

No, `(val) < INT_MIN` is always true for `unsigned` val. So, the check against 0 is needed.

Reply:
> Also, please fix the commit title since it is no longer in `mpl`

fixed.

Reply:
> > Refer to [#6145 (comment)](https://github.com/pmodels/mpich/pull/6145#discussion_r966589870)
> 
> No, `(val) < INT_MIN` is always true for `unsigned` val. So, the check against 0 is needed.

Alright. I see what is happening.

Reply:
BTW, what is `assert(0), 0` when `NDEBUG` is defined?

Reply:
When NDEBUG not defined, assert is a no-op.